overlord/devicestate/devicemgr.go

```diff
@@ -183,9 +183,26 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, runner *state.T
 	boot.RunFDESetupHook = m.runFDESetupHook
 	hookManager.Register(regexp.MustCompile("^fde-setup$"), newFdeSetupHandler)
 
+	// ensure permissions are ok
+	if err := ensureFileDirPermissions(); err != nil {
```

Comment:
Hm here, or in StartUp(). As a precedent, we already have some code that sets up /var/lib/snapd/save mount in startup. Also StartUp() runs some early config code thus mutating the device state, so perhaps it'd be better to move the void fixup there too, such that Manager() is mostly about reading modeenv and registering various hooks.

Reply:
Updated